catalyst_voices/packages/catalyst_voices_services/lib/src/user/user_service.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:async';
+
+import 'package:catalyst_cardano_serialization/catalyst_cardano_serialization.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_services/catalyst_voices_services.dart';
+import 'package:logging/logging.dart';
+
+abstract interface class UserService {
+  Account? get account;
+
+  Stream<Account?> get watchAccount;
+
+  Keychain? get keychain;
+
+  Future<List<Keychain>> get keychains;
+
+  Stream<Keychain?> get watchKeychain;
+
+  Future<void> useActiveAccount();
+
+  Future<void> removeActiveAccount();
+
+  Future<void> switchToAccount(Account account);
+
+  Future<void> switchToKeychain(String id);
+
+  Future<void> dispose();
+}
+
+final class UserServiceImpl implements UserService {
+  final KeychainProvider _keychainProvider;
+  final UserStorage _userStorage;
+
+  User? _user;
+  final _userSC = StreamController<User?>.broadcast();
+
+  Keychain? _keychain;
+  final _keychainSC = StreamController<Keychain?>.broadcast();
+  StreamSubscription<bool>? _keychainUnlockSub;
+
+  final _logger = Logger('UserService');
```

Comment:
I would put it at the top of the class. In general we're following dart code style, so first we put static fields, then static vars, then final fields, then variables. I understand that for `_user` and `_userSC` breaking above order is beneficial then breaking these for the logger doesn't add any value